skyportal/tests/frontend/test_gcnevents.py

```diff
@@ -53,9 +55,27 @@ def test_gcnevents(
     status, data = api("GET", f"sources/{obj_id}", token=view_only_token)
     assert status == 200
 
+    galaxy_name = str(uuid.uuid4())
+    data = {
+        'catalog_name': 'galaxy_in_Fermi',
+        'catalog_data': {'name': [galaxy_name], 'ra': [228.5], 'dec': [35.5]},
+    }
+
+    status, data = api('POST', 'galaxy_catalog', data=data, token=super_admin_token)
+    assert status == 200
+    assert data['status'] == 'success'
+
+    # wait for galaxies to load
+    time.sleep(15)
```

Comment:
Is there a way to issue a query to see if the galaxies have loaded and then do a back off if need be before retry (up to a certain number of tries)? Hardcoding in a sleep seems inefficient especially since these tests can be run on machines of different speeds.

Reply:
@profjsb added a while loop, see if you like it?@profjsb added a while loop, see if you like it?